code/src/reducers/labyrinth.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable linebreak-style */
+import { createSlice } from '@reduxjs/toolkit'
+import { v4 as uuidv4 } from 'uuid';
+
+const initialState = {
+  userName: '',
+  name: '',
+  description: '',
+  direction: '',
+  moves: '',
+  history: [],
+  loading: false,
+  coordinates: ''
+}
+
+export const labyrinth = createSlice({
+  name: 'labyrinth',
+  initialState,
+  reducers: {
+
+    setUserName: (store, action) => {
+      const { input } = action.payload
+      const userId = uuidv4(input);
```

Comment:
As kaja mention, really liked that you used the uniq id for using the same username!